source_code/aux_mcu_v4/src/main.c

```diff
@@ -277,7 +289,7 @@ void main_init_stack_tracking(void)
     uint32_t stack_end;
     uint32_t stack_size;
 
-    stack_start = (uint32_t) &_estack;
+    stack_start = main_get_SP() - sizeof(uint32_t);
     stack_end = (uint32_t) &_sstack;
     stack_size = stack_start - stack_end;
```

Comment:
is the memset call below an inline? if not, would that still work? :)

Reply:
Good point. Let me fix that. :-)

Reply:
Actually, its not an issue in the current code. Call to memset and memset itself does not use the stack. However, fixing it for futureproofing.